docs/config-file.md

```diff
@@ -18,11 +18,9 @@ This function must return an object with the following shape:
 export default async function () {
   return {
     glee: {},
-    kafka: {},
-    websocket: {},
-    mqtt: {},
-    cluster: {},
-    http: {}
+    docs: {},
+    server: {},
+    cluster: {}
   }
 }
```

Comment:
[This](https://github.com/asyncapi/glee/blob/b0c1670f777ad2a4dd0e33004d5ab78a72964463/docs/config-file.md) was the old format for `glee-config` and it was done to configure WebSocket servers for socket-io or pass in a custom server object. So building on that I am trying to add a new object named `server` where we would pass in the custom server config. The parser would take a server name from the spec and see if any config exists in the config file and then load that accordingly.

Reply:
It looks good, the only thing that worries me a little about this approach is writing an interface for it. I mean we want type suggestion/safety on the file right? with server name as a key I am not sure how we can write protocol specific interfaces. 🤔

Reply:
I think we can still achieve that, I am not sure but I used notion API which had something like this kind of type 

```ts
interface T {
  server: {
    [name:string]: {
      httpAdapter: any,
      adapter: string,
      port: number
    }
  }
}

```

Reply:
yeah, If type can be checked then let's do this. 👍